src/System.Private.Xml/tests/Writers/XmlWriterApi/TCErrorState.cs

```diff
@@ -1,87 +1,301 @@
+                try
+                {
+                    w.WriteStartDocument();
+                    w.WriteStartElement("Root");
+                    w.WriteEndElement();
+                    w.WriteEntityRef("ent");
+                }
+                catch (InvalidOperationException e)
+                {
+                    CError.WriteLineIgnore("Exception: " + e.ToString());
+                    CError.Compare(w.WriteState, WriteState.Error, "WriteState should be Error");
+                    return;
+                }
             }
+            CError.WriteLine("Did not throw exception");
+            Assert.True(false);
+        }
 
-
-            // for function state_3
+        //[Variation(id = 3, Desc = "CharEntity after Document should error - PROLOG", Pri = 1)]
+        [Theory]
+        [XmlWriterInlineData]
+        public void state_3(XmlWriterUtils utils)
+        {
+            using (XmlWriter w = utils.CreateWriter())
             {
-                this.AddChild(new CVariation(state_3) { Attribute = new Variation("CharEntity after Document should error - PROLOG") { id = 3, Pri = 1 } });
+                try
+                {
+                    w.WriteStartDocument();
+                    w.WriteCharEntity('\uD23E');
+                }
+                catch (InvalidOperationException e)
+                {
+                    CError.WriteLineIgnore("Exception: " + e.ToString());
+                    CError.Compare(w.WriteState, WriteState.Error, "WriteState should be Error");
+                    return;
+                }
             }
+            CError.WriteLine("Did not throw exception");
+            Assert.True(false);
+        }
 
-
-            // for function state_4
+        //[Variation(id = 4, Desc = "CharEntity after Document should error - EPILOG", Pri = 1)]
+        [Theory]
+        [XmlWriterInlineData]
+        public void state_4(XmlWriterUtils utils)
+        {
+            using (XmlWriter w = utils.CreateWriter())
             {
-                this.AddChild(new CVariation(state_4) { Attribute = new Variation("CharEntity after Document should error - EPILOG") { id = 4, Pri = 1 } });
+                try
+                {
+                    w.WriteStartDocument();
+                    w.WriteStartElement("Root");
+                    w.WriteEndElement();
+                    w.WriteCharEntity('\uD23E');
+                }
+                catch (InvalidOperationException e)
+                {
+                    CError.WriteLineIgnore("Exception: " + e.ToString());
+                    CError.Compare(w.WriteState, WriteState.Error, "WriteState should be Error");
+                    return;
+                }
             }
+            CError.WriteLine("Did not throw exception");
+            Assert.True(false);
+        }
 
-
-            // for function state_5
+        //[Variation(id = 5, Desc = "SurrogateCharEntity after Document should error - PROLOG", Pri = 1)]
```

Comment:
@danmosemsft leaving a text sounds good, some regexes are tricky though since the description occasionally contains quotes - //[Variation.. should actually all be gone right now since I've converted them to //[XmlWriterInlineData - you are looking at the intermediate change